tests/src/test/java/io/openshift/booster/OpenshiftIT.java

```diff
@@ -0,0 +1,168 @@
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package io.openshift.booster;
+
+import static com.jayway.awaitility.Awaitility.await;
+import static com.jayway.restassured.RestAssured.get;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.core.IsEqual.equalTo;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import com.jayway.restassured.RestAssured;
+import com.jayway.restassured.response.Response;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.openshift.api.model.DeploymentConfig;
+import io.fabric8.openshift.api.model.Route;
+import io.openshift.booster.test.OpenShiftTestAssistant;
+
+/**
+ * @author Martin Kouba
+ */
+public class OpenshiftIT {
+
+    private static final String NAME_SERVICE_APP = "wfswarm-circuit-breaker-name";
+    private static final String GREETING_SERVICE_APP = "wfswarm-circuit-breaker-greeting";
+
+    private static final String NAME_OK = "{\"state\":\"ok\"}";
+    private static final String NAME_FAIL = "{\"state\":\"fail\"}";
+    private static final String CB_CLOSED = "\"state\":\"closed\"";
+    private static final String CB_OPEN = "\"state\":\"open\"";
+    private static final String HELLO_OK = "Hello, World!";
+    private static final String HELLO_FALLBACK = "Hello, Fallback!";
+
+    // See also circuitBreaker.sleepWindowInMilliseconds
+    private static final long SLEEP_WINDOW = 5000l;
+    // See also circuitBreaker.requestVolumeThreshold
+    private static final long REQUEST_THRESHOLD = 3;
+
+    private static final OpenShiftTestAssistant OPENSHIFT = new OpenShiftTestAssistant();
+
+    private static String nameBaseUri;
+
+    private static String greetingBaseUri;
+
+    @BeforeClass
+    public static void setup() throws Exception {
+
+        nameBaseUri = deployApp(NAME_SERVICE_APP, System.getProperty("nameServiceTemplate"));
+        greetingBaseUri = deployApp(GREETING_SERVICE_APP, System.getProperty("greetingServiceTemplate"));
+
+        await().atMost(5, TimeUnit.MINUTES).until(() -> {
+            List<Pod> list = OPENSHIFT.client().pods().inNamespace(OPENSHIFT.project()).list().getItems();
+            return list.stream()
+                    .filter(pod -> pod.getMetadata().getName().startsWith(NAME_SERVICE_APP) || pod.getMetadata().getName().startsWith(GREETING_SERVICE_APP))
+                    .filter(pod -> "running".equalsIgnoreCase(pod.getStatus().getPhase())).collect(Collectors.toList()).size() >= 2;
+        });
+
+        System.out.println("Pods running, waiting for readiness probe...");
```

Comment:
Do we really need to wait for the pods to become "running" before we start waiting for the successful ping response? Maybe we can just wait for the ping directly.